### PR TITLE
Va iir 688 further support cleanup

### PR DESCRIPTION
**Pull request by AdamKing0126**
## Summary

I'm Adam King, I work on the VA-IIR team, and we own this component.

This work is part of [GitHub Issue #688](https://github.com/department-of-veterans-affairs/va-iir/issues/688) to decommission the /v0/post911_gi_bill_status endpoint. Because of the large number of changes, I'm breaking it up into multiple smaller, more digestible PRs.

This PR includes some additional cleanup, before the removal of the production code:
- Updates to Swagger documentation for minor data contract changes
- Test updates to test the v1 endpoint, remove tests for the v0 endpoint
- Other small housekeeping items.

Once this PR, along with 16944 and 16945 are merged, there will be a final PR to remove the v0 controller, its associated helper models, and the entry in `config/routes.rb`

- [#16944](https://github.com/department-of-veterans-affairs/vets-api/pull/16944) - ~removing tests for the v0 controller~ MERGED
- [#16945](https://github.com/department-of-veterans-affairs/vets-api/pull/16945) - ~removing response and service spec files, along with recorded VCR cassettes~ MERGED
- [#16955](https://github.com/department-of-veterans-affairs/vets-api/pull/16955) (still in "Draft" status) - actual removal of production code: `v0` controller, associated data models, entry in `config/routes.rb`

### CODE REVIEW

File: app/swagger/swagger/v1/requests/post911_gi_bill_statuses.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ class Swagger::V1::Requests::Post911GIBillStatuses
     property :end_date, type: %i[string null], example: '2012-12-01T05:00:00.000+00:00'
     property :facility_code, type: %i[string null], example: '12345678'
     property :facility_name, type: %i[string null], example: 'Purdue University'
-    property :participant_id, type: %i[string null], example: '11170323'
+    property :participant_id, type: %i[integer null], example: '11170323'
```
Comment (stevelong00) [inline app/swagger/swagger/v1/requests/post911_gi_bill_statuses.rb:86]: Suggest removing quotes from the example now that this is an integer

**Review by stevelong00 — DISMISSED** [review]
Looks good.  One suggested edit.

File: app/swagger/swagger/v1/requests/post911_gi_bill_statuses.rb (hunk 1 above)
Comment (AdamKing0126) [reply app/swagger/swagger/v1/requests/post911_gi_bill_statuses.rb]: 👌 my editor kept formatting the number strangely (no idea why), so I changed the example to `1234` but I *did* remove the quotes.  Thanks!

**Review by stevelong00 — DISMISSED** [review]
Great!

**Review by stevelong00 — APPROVED** [review]
Looks good!

File: spec/requests/swagger_spec.rb [hunk 2]
```diff
@@ -3436,7 +3415,38 @@
       subject.untested_mappings.delete('/v0/sign_in/callback')
       subject.untested_mappings.delete('/v0/sign_in/logout')
 
+      subject.untested_mappings.delete('/v0/post911_gi_bill_status')
```
Comment (AdamKing0126) [inline spec/requests/swagger_spec.rb:3418]: this entry will be deleted once we remove the endpoint and update swagger documentation

**Review by ericboehs — APPROVED** [review]

**Pull request merged**